pkg/apis/crds/karpenter.k8s.aws_ec2nodeclasses.yaml

```diff
@@ -566,6 +577,18 @@ spec:
                   rule: (has(self.role) && !has(self.instanceProfile)) || (!has(self.role) && has(self.instanceProfile))
                 - message: changing from 'instanceProfile' to 'role' is not supported. You must delete and recreate this node class if you want to change this.
                   rule: (has(oldSelf.role) && has(self.role)) || (has(oldSelf.instanceProfile) && has(self.instanceProfile))
+                - message: if set, amiFamily must be 'AL2' or 'Custom' when using an AL2 alias
+                  rule: '!has(self.amiFamily) || (self.amiSelectorTerms.exists(x, has(x.alias) && x.alias.find(''^[^@]+'') == ''al2'') ? (self.amiFamily == ''Custom'' || self.amiFamily == ''AL2'') : true)'
```

Comment:
This regex captures any character before the ampersand, which captures the entire AMI family section of the alias. You could specify that this should only be alphanumeric characters, but it isn't really necessary since that's validated by other rules.

Reply:
nit: ampersand is & not @ :P 

Seems we should just make sure it's alphanumeric, because we can, no?